standards/commits.md

```diff
@@ -1,52 +1,72 @@
-            _Now you have 1 commit, dedicated to the **app.component** scope._  
-    
-    * Next, `git add` the login files and create a new commit message for these files.   
-        > `git commit -m "feat(login): create/setup"`  
-            _Note that this message is less verbose because it isnt necessary to add detail_
+# Bitwise's convention on how to structure, write, and use commit messages
+
+Using a convention for commit messages has several benefits. For one, it makes it easier to find individual commits. It also forces us to keep our commits small enough to be able easily reason what they're doing. And lastly to be able to revert small chunks of code if needed.
+
+## Git Commit Messages Conventions
+
+There are [many conventions](https://github.com/conventional-changelog/commitlint#shared-configuration) available, many follow a close resemblance to the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/).
+
+👍 **The one we use at Bitwise is based on the [Karma Runner](http://karma-runner.github.io/1.0/dev/git-commit-msg.html) project.** Please review it.
+
+## Guidelines:
+
+- Each commit should contain files that all pertain to the same _scope_
+- If you have changed multiple files that apply to different scopes, add and commit them separately
+- Make your messages as direct and succinct as possible
+
+## Format
+
+### Your commit messages should be formatted as follows:
+
+`type(scope):subject`
+
+Commit **Type**: Select one of the following 8 commit types. These are your _only_ options for commit type:
+
+- **_feat_**: A new feature for the application user. Rolling out a new module, new piece of functionality, etc.
+- **_fix_**: Bug fix to your production code. Dealing with GH Issues, fixing a bug, etc.
+- **_docs_**: Changes to your documentation. Adding a comment, editing a comment, changing README.md files, etc.
+- **_style_**: Changes to your code formatting. _This does **not** address CSS styling_ but rather the style of your code. Reformatting, adding semi-colons, etc.
+- **_refactor_**: Refactor to production code. Upgrading a package and changing your code to meet the new demands, changing a **let** or a **const** name, etc.
+- **_test_**: All things that apply to unit testing. Creating tests, refactoring tests, etc. No changes to production code occur.
+- **_chore_**: Updating gulp, webpack, package.json files. This is developer-facing _only_.
+- **_workaround_**: Temporary fix until a more robust solution is found or until other factors are resolved.
+
+Commit **Scope**: Narrow the scope of the commit to a one or two word description in parentheses
+
+Message **Subject**: Favor imperative mood, present tense, active voice, and start with verbs. Don't use a period at the end. Think of it as a newspaper headline.
+
+Message **Body**: If necessary, write a short sentence detailing pertinent information for other developers. This is normally unnecessary but some use cases are:
```

Comment:
Should this `Message **Body**` be updated to `Message **Subject**`? I like the information provided, just noticed that the example commit above references it as subject.

Reply:
hi @gtrujillo15 a git commit message is usually composed of a subject and body.

You can either do:

`git commit -m "this is a subject"` - this only takes a subject

`git commit` this will open an editor such as vim. First line is the subject and any lines afterwards is considered the body. For example:

```
this is a subject

hello world I did this because of x and y. This is the message body.
```

But maybe I am misunderstanding your suggestion? 🤔 If so, can you please clarify. Thanks!

Reply:
Ohh, I see. I just meant that on line 18 the commit message format was changed from `type(scope):body` to `type(scope):subject` (line 21), so I figured the term was changed from 'body' to 'subject'. Hopefully I make more sense! :smile:

Reply:
@gtrujillo15 I see now. However, it was referring to body. I have updated the PR with examples in doc to hopefully clarify it a bit more. Please see PR, you can preview markdown here: https://github.com/amerikan/standards-and-practices/blob/main/standards/commits.md

Reply:
That helps so much! Thank you for the clarification!